samcli/lib/deploy/deployer.py

```diff
@@ -345,9 +345,8 @@ def describe_stack_events(self, stack_name, time_stamp_marker, **kwargs):
 
         while stack_change_in_progress and retry_attempts <= self.max_attempts:
             try:
-
                 # Only sleep if there have been no retry_attempts
-                time.sleep(self.client_sleep if retry_attempts == 0 else 0)
+                time.sleep(0 if retry_attempts else self.client_sleep)
```

Comment:
What was the complaint here? As above, not sure why it prefers boolean logic on integers.